packages/x-data-grid-generator/src/hooks/serverUtils.ts

```diff
@@ -261,6 +285,41 @@ const getFilteredRows = (
   );
 };
 
+const applyAggregation = (
+  aggregationModel: GridAggregationModel,
+  colDefs: GridColDef[],
+  rows: GridRowModel[],
+  groupId: string = 'root',
+) => {
+  const columnsToAggregate = Object.keys(aggregationModel);
+  if (columnsToAggregate.length === 0) {
+    return {};
+  }
+
+  const aggregateValues: GridValidRowModel = {};
+  columnsToAggregate.forEach((field) => {
+    const type = colDefs.find(({ field: f }) => f === field)?.type;
+    if (!type) {
+      return;
+    }
+    const availableAggregationFunctions = getAvailableAggregationFunctions(type);
+    if (!availableAggregationFunctions.has(aggregationModel[field])) {
+      return;
+    }
+    const aggregationFunction = availableAggregationFunctions.get(aggregationModel[field]);
+    if (!aggregationFunction) {
+      return;
+    }
+    const values = rows.map((row) => row[field]);
+    aggregateValues[`${field}Aggregate`] = aggregationFunction.apply({
```

Comment:
To make the response cleaner, what do you think of having all aggregation data under `aggregation` object.
This is implementation detail anyway, but our examples would be cleaner.
So, in the response you would have
```ts
{
  rows: [
    {
      ...
      // group aggregation
      aggregation: {
        age: 30,
      }
    },
    ...
  ],
  // total
  aggregation: {
    age: 27,
  },
}
```

Reply:
It looks better, though I think having a non-nested value structure has it's benefits too, for example one quick one would be, we won't have to have different conditions for root and nested groups in `getAggregatedValue`

Let's look at some pseudo code:

**1. Current structure**
```tsx
const response = {
  rows: [
    {
      ...
      ageAggregate: 30,
    },
    ...
  ],
  // total
  aggregateRow: {
    ageAggregate: 27,
  },
}

const dataSource = {
  getRows: (...params) => { ...logic },
  getAggregatedValue: (row, field) => {
    return row[`${field}Aggregate`];
  }
}
```

**2. New structure**
```tsx
const response = {
  rows: [
    {
      ...
      // group aggregation
      aggregation: {
        age: 30,
      }
    },
    ...
  ],
  // total
  aggregation: {
    age: 27,
  },
}

const dataSource = {
  getRows: (...params) => { ...logic },
  getAggregatedValue: (row, field) => {
    if (isRootRow) { // the user need some extra information to check if the row is root row
      return row[field];
    } else {
      return row.aggregation?.[field];
    }
  }
}
```

It's not a big deal though, we could pass a third param as `groupId` which users can use like
```js
if (groupId === GRID_ROOT_GROUP_ID) {
...
```
However I'm a bit more inclined on keeping it simple.

Does this argument feel convincing?

Reply:
About the following key naming change:

```diff
const getRowsResponse = {
  rows: [...],
  rowCount: 100,
- aggregateRow: { ... },
+ aggregation: { ... },
}
```

I named it `aggregateRow`, as it's a bit more self-explanatory to me and also related to the `rows` and `rowCount` keys. This perception might be different for others but I feel the keyword `aggregation` gives a more abstract gesture and one have to think twice to figure out it's about the footer aggregation row. Do you feel differently?

Reply:
>However I'm a bit more inclined on keeping it simple.

This is also fine. As I mentioned _This is implementation detail anyway_, so I don't have a strong opinion either side.
Developers can change this in any way they like. Let's keep `getAggregatedValue` simple then.

Regarding the naming, for the API it makes more sense not to mention row, since this is UI detail. The values are simply an aggregation. You already pointed out that having `Row` suffix makes it more aligned with the rest of the interface, so I am also fine to keep it like this for now, but I plan to propose a change regarding this and make the interface more generic.

In the future, one API can serve multiple X components (grid and charts for example) and having `row` in the name will create a problem.